### DISCUSSION TITLE
Escaping from "docker mc attach"

### DISCUSSION THREAD

**Discussion by breeegz** [itzg/docker-minecraft-server#1388] [Q&A] [answered]
I see in the README that if I perform a Control-P Control-Q I can detach from the container after attaching.  I'm not having much luck, and I'm sure that I'm just doing something simple.  Control-C kills my terminal and kills the container, not the desired effect.

So far, I think I've done it correctly one time, but I struggle with it each time I try detaching.  My "workaround" is to leave a terminal connected to the container until I need to reboot.  But if someone can clarify that portion of the documentation, then maybe it will help me or someone else doing the same "wrong" keystrokes.  I'm hesitant to call this a bug since I think it's user error.

Docker is running on my Ubuntu 20.04 desktop, FWIW.

**Comment by itzg** [comment, marked as answer]
What more is there to document? This is not specific to this image, so you can read more at https://docs.docker.com/engine/reference/commandline/attach/

Besides, do the other options to send commands not work in your scenario? `docker attach` is considered a last resort.

**Reply by breeegz** [reply]
> What more is there to document? 

Yeah, I was pretty sure it was on my end.  I'm just getting familiar with Containers and Docker.  

>This is not specific to this image, so you can read more at https://docs.docker.com/engine/reference/commandline/attach/

I didn't think to try searching on Docker docs.  I think that will probably help in this instance...  like I told my first girlfriend, it's not you, it's me..  lol
 
> Do the other options to send commands not work in your scenario? `docker attach` is considered a last resort.

`docker exec -i mc rcon-cli` works fine
`docker exec -i mc rcon-cli [command]` works fine

I didn't realize "docker attach mc" was considered a last resort.  I was using it to watch the activity on my container.  Container Initialization, Chat, deaths, Server Crashes, etc.  If the standard output goes to a log, I could "tail -f" it and get my desired/expected outcome without using it.   Like I alluded to in my original question, I kinda feel like a newb for even asking..  I appreciate your response, and I hope that it helps someone else in the future.

Mid-reply update:
`docker logs -f mc` works better than `docker attach mc`.  I can close the terminal or control-c and it doesn't kill the container with SIGKILL.  It gives me all the bootup and shutdown text that I was watching with docker attach.  I can't enter commands from there, but that's exactly the same limitaion with docker attach.  If you have any other suggestions, then I'll listen, but for now I think this discussion can be closed  and marked as answered.

**Reply by itzg** [reply]
Excellent, yes, hearing your specific use case, then `logs` is indeed the way to go.
